core/src/client/mod.rs

```diff
@@ -675,4 +703,540 @@ impl StarknetClient for StarknetClientImpl {
 
         Ok(Address::from(evm_address_sliced))
     }
+
+    async fn starknet_tx_into_eth_tx(
+        &self,
+        tx: StarknetTransaction,
+        block_hash: Option<PrimitiveH256>,
+        block_number: Option<U256>,
+    ) -> Result<EtherTransaction, KakarotClientError> {
+        let mut ether_tx = EtherTransaction::default();
+        let class_hash;
+
+        match tx {
+            StarknetTransaction::Invoke(invoke_tx) => {
+                match invoke_tx {
+                    InvokeTransaction::V0(v0) => {
+                        // Extract relevant fields from InvokeTransactionV0 and convert them to the corresponding fields in EtherTransaction
+                        ether_tx.hash =
```

Comment:
why are we assigning the class hash of the account to the ether_tx.hash, which is the transaction hash?

Reply:
We are not, we have another "class_hash" right below it